libsel4vmmplatsupport/src/arch/arm/psci.c

```diff
@@ -50,15 +50,24 @@ int handle_psci(vm_vcpu_t *vcpu, seL4_Word fn_number, bool convention)
         smc_set_return_value(&regs, 0x00010000); /* version 1 */
         break;
     case PSCI_CPU_ON: {
-        uintptr_t target_cpu = smc_get_arg(&regs, 1);
+        uintptr_t requested_cpu = smc_get_arg(&regs, 1);
```

Comment:
I think the guest would still pass MPIDR here, and that is not a linear number if there are multiple clusters to emulate. So we cannot use this as array index, but have to look it up in all vCPUs the guest got. We should really call even the variable `target_vmpidr` then.